demo/core_13_$.utils.code.js

```diff
@@ -136,3 +136,60 @@ $.utils.code.toSourceSafe = function(value) {
     throw e;
   }
 };
+
+$.utils.code.formatForEval = function(src) {
+  // Eval treats {} as an empty block (return value undefined).
+  // Eval treats {'a': 1} as a syntax error.
+  // Eval treats {a: 1} as block with a labeled statement (return value 1).
+  // Detect these cases and enclose in parenthesis.
+  // But don't mess with: {var x = 1; x + x;}
+  // This is consistent with the console on Chrome and Node.
+  try {
+    var ast = $.utils.acorn.parse(src, { ecmaVersion: 5 });
+  } catch (e) {
+    try {
+      ast = $.utils.acorn.parseExpressionAt(src, 0, { ecmaVersion: 5 });
+      if (src.substring(ast.end).trim() !== '') {
+        throw SyntaxError('Not just an expression');
```

Comment:
Does this reject trailing comments on object literals?  Maybe add some test cases for that, to be sure.

Reply:
Heh, good catch.  Comments don't show up in the AST at all.  So we got back:
`({} // comment)`
which is not valid.  Changed to always add a line break before the closing paren.